feathery-org/feathery-react
# Stripe Integration

This is V1 of stripe integration and supports payment method(s) setup only.
See functional requirements: https://docs.google.com/document/d/1etw2-A-zjceqvqJmNJYBKNSX6unxCf3Gnd5jVbyqS8I/edit?usp=sharing
See Tech Integration: https://docs.google.com/document/d/1vqrKkI-iF4se0JcBQjxyLl8vb_-yxpC3lIqIe0we3-k/edit?usp=sharing

Test Outline:

- Optional PM field and no card entered - does NOT setup
- Optional PM field and card entered - does setup
- Required PM field and no card entered - required field error
- Required PM field and card entered - does setup
- Card entry incomplete - cannot submit - error shown
- Test live form and review and make sure the proper stripe key is used.  Simulate using two stripe accounts.  Use hosted forms with diff feathery api keys.  Using 2 stripe accounts ([eric.pias@comcast.net](mailto:eric.pias@comcast.net) (PW saved in Safari) and [ericdpias@gmail.com](mailto:ericdpias@gmail.com) (FF)).
- Multi-step form
- Setup card and then go back, must setup PM again if required or card entered.
- Verify card not setup on steps where card does not appear.
- Test multiple cards  (must both go to same customer)
- Two cards one step
- Two steps, each with a PM field
- Two steps, first 2 cards, 2nd 1 card.
- Test a form with no PM field
- Test DB master against stripe Lib and BE.
